compiler/src/codegen/compcore.re

```diff
@@ -33,7 +33,7 @@ let gensym_label = s => {
 };
 
 /* Number of swap variables to allocate */
-let swap_slots_i32 = [|Type.int32, Type.int32|];
+let swap_slots_i32 = [|Type.int32, Type.int32, Type.int32|];
```

Comment:
The beauty of optimizations means that we can have as many swap slots as we want, and Binaryen will throw out the unused ones. I doubt we'll ever need more than three, though.